scylla/tests/integration/batch_statements.rs

```diff
@@ -0,0 +1,403 @@
+        .collect();
+    assert_eq!(
+        rows.len(),
+        BATCH_COUNT,
+        "Expected {} rows, got {}",
+        BATCH_COUNT,
+        rows.len()
+    );
+    for (k0, k1, v) in rows {
+        assert_eq!(k0, test_name);
+        assert_eq!(v, if k1 % 20 == 0 { k1 + 1 } else { k1 + 2 });
+    }
+}
+
+#[tokio::test]
+async fn test_batch_of_bound_statements_named_variables() {
+    setup_tracing();
+    let test_name = String::from("test_batch_bound_statements_named_variables");
+    let (session, ks) = create_test_session(&test_name).await;
+
+    let query_str = format!(
+        "INSERT INTO {}.{} (k0, k1, v) VALUES (:k0, :k1, :v)",
+        ks, &test_name
+    );
+    let prepared = session.prepare(query_str).await.unwrap();
+
+    let mut batch = Batch::new(BatchType::Unlogged);
+    let mut batch_values = Vec::with_capacity(BATCH_COUNT);
+    for i in 0..BATCH_COUNT as i32 {
+        batch.append_statement(prepared.clone());
+        let mut values = HashMap::new();
+        values.insert("k0", CqlValue::Text(test_name.clone()));
+        values.insert("k1", CqlValue::Int(i));
+        values.insert("v", CqlValue::Int(i + 1));
+        batch_values.push(values);
+    }
+    session.batch(&batch, batch_values).await.unwrap();
+    verify_batch_insert(&session, &ks, &test_name, BATCH_COUNT).await;
+}
+
+#[tokio::test]
+async fn test_batch_of_mixed_bound_and_simple_statements() {
+    setup_tracing();
+    let test_name = String::from("test_batch_mixed_bound_and_simple_statements");
+    let (session, ks) = create_test_session(&test_name).await;
+
+    let query_str = format!(
+        "INSERT INTO {}.{} (k0, k1, v) VALUES (?, ?, ?)",
+        ks, &test_name
+    );
+    let prepared_bound = session
+        .prepare(Query::new(query_str.clone()))
+        .await
+        .unwrap();
+
+    let mut batch = Batch::new(BatchType::Unlogged);
+    let mut batch_values = Vec::with_capacity(BATCH_COUNT);
+    for i in 0..BATCH_COUNT as i32 {
+        if i % 2 == 1 {
+            let simple_statement = Query::new(format!(
+                "INSERT INTO {}.{} (k0, k1, v) VALUES ('{}', {}, {})",
+                ks,
+                &test_name,
+                &test_name,
+                i,
+                i + 1
+            ));
+            batch.append_statement(simple_statement);
+            batch_values.push(vec![]);
+        } else {
+            batch.append_statement(prepared_bound.clone());
+            batch_values.push(vec![
+                CqlValue::Text(test_name.clone()),
+                CqlValue::Int(i),
+                CqlValue::Int(i + 1),
+            ]);
+        }
+    }
+    session.batch(&batch, batch_values).await.unwrap();
+    verify_batch_insert(&session, &ks, &test_name, BATCH_COUNT).await;
+}
+
+/// TODO: Remove #[ignore] once LWTs are supported with tablets.
+#[tokio::test]
+#[ignore]
+async fn test_cas_batch() {
+    setup_tracing();
+    let test_name = String::from("test_cas_batch");
```

Comment:
I guess you got the inspiration from `test_lwt_optimization_works_with_tablets`. The difference is that this test does not depend on tablets, while `test_lwt_optimization_works_with_tablets` does. 
Please disable tablets explicitly during keyspace creation. This  will allow the test to actually run.